custom_components/button_plus/buttonplushub.py

```diff
@@ -42,17 +43,72 @@ def __init__(self, hass: HomeAssistant, config: DeviceConfiguration, entry: Conf
             suggested_area=self.config.core.location,
             name=self._name,
             model="Base Module",
-            hw_version=config.info.firmware
+            sw_version=config.info.firmware
+        )
+
+        # 1 or none display module
+        self.display_module = next((self.create_display_module(hass, entry, self) for _ in self.connector(ConnectorEnum.DISPLAY)), None)
+        self.display_bar = [(connector_id, self.create_bar_module(hass, entry, self, connector_id)) for connector_id in self.connector(ConnectorEnum.BAR)]
+
+    def create_display_module(self, hass: HomeAssistant, entry: ConfigEntry, hub: ButtonPlusHub) -> None:
+        _LOGGER.debug(f"Add display module from '{hub.hub_id}'")
+        device_registry = dr.async_get(hass)
+
+        device = device_registry.async_get_or_create(
+            config_entry_id=entry.entry_id,
+            # connections={(DOMAIN, hub.config.info.device_id)},
+            name=f"{hub.name} Display Module",
+            model="Display Module",
+            manufacturer=MANUFACTURER,
+            suggested_area=hub.config.core.location,
+            identifiers={
+                (DOMAIN, f"{hub.hub_id} Display Module")
+            },
+            via_device=(DOMAIN, hub.hub_id)
         )
+        
+        return device
+
+
+    def create_bar_module(self, hass: HomeAssistant, entry: ConfigEntry, hub: ButtonPlusHub, connector_id: int) -> None:
+        _LOGGER.debug(f"Add bar module from '{hub.hub_id}' with connector '{connector_id}'")
+        device_registry = dr.async_get(hass)
+
+        device = device_registry.async_get_or_create(
+            config_entry_id=entry.entry_id,
+            # connections={(DOMAIN, hub.config.info.device_id)},
+            name=f"{hub._name} BAR Module {connector_id}",
+            model="Bar module",
+            manufacturer=MANUFACTURER,
+            suggested_area=hub.config.core.location,
+            identifiers={
+                (DOMAIN, f"{hub.hub_id} BAR Module {connector_id}")
+            },
+            via_device=(DOMAIN, hub.hub_id)
+        )
+
+        return device
+
+    def connector(self, connector_type: ConnectorEnum):
+        return [
+            connector.connector_id
+            for connector in self.config.info.connectors
+            if connector.connector_type_enum() in [connector_type]
+        ]
+
 
     @property
-    def client(self) -> LocalApiClient:
+    def client(self): # -> LocalApiClient:
```

Comment:
Does this client no longer return the `LocalApiClient`? I don't see why this typing should be removed.

Reply:
Changed this back